File: infiniAngular-main/User/src/model/BankAccount.ts
 
import { User } from './User';

import { Transaction } from './Transaction';
import { typeAccount } from './TypeAccount';
 
export class BankAccount {
    rib: number;
    balance: number;
    openDate: Date;
    code: number;
    loyaltyScore: number;
    typeAccount: typeAccount;
    user: number;
    
   
    constructor() {
        this.rib = 0;
        this.balance = 0;
        this.openDate = new Date();
        this.code = 0;
        this.loyaltyScore = 0;
        this.typeAccount = typeAccount.CURRENT;
        this.user = 0;
    }

    
    }

File: MFB/src/main/java/tn/esprit/mfb/entity/BankAccount.java
package tn.esprit.mfb.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import tn.esprit.mfb.model.TypeAccount;

import java.time.LocalDate;
import java.util.Set;


@Entity
@Getter
@Setter
public class BankAccount {

    @Id
    @Column(nullable = false, updatable = false)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long rib;

    @Column
    private Double balance;

    @Column
    private LocalDate openDate;

    @Column
    private Integer code;

    @Column
    private Integer loyaltyScore;

    @Column
    @Enumerated(EnumType.STRING)
    private TypeAccount typeAccount;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;

    @OneToMany(mappedBy = "sourceRIB")
    private Set<Transaction> transactions;

    public void addFunds(double amount) {
        this.balance += amount;
    }



}